packages/core/src/routes/connector.ts

```diff
@@ -132,18 +132,35 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
     '/connectors/:id',
     koaGuard({
       params: object({ id: string().min(1) }),
-      body: Connectors.createGuard.omit({ id: true, enabled: true, createdAt: true }).partial(),
+      body: Connectors.createGuard.pick({ config: true, metadata: true }).partial(),
     }),
+
     async (ctx, next) => {
       const {
         params: { id },
+        body: { config, metadata: databaseMetadata },
         body,
       } = ctx.guard;
 
-      const { metadata, type, validateConfig } = await getLogtoConnectorById(id);
+      const {
+        metadata: { isStandard },
+        metadata,
+        type,
+        validateConfig,
+      } = await getLogtoConnectorById(id);
+
+      if (config) {
+        validateConfig(config);
+      }
 
-      if (body.config) {
-        validateConfig(body.config);
+      if (isStandard === true) {
```

Comment:
for a PATCH endpoint, the user don't have to set `metadata`, consider add an if statement